src/map/skill.c

```diff
@@ -13048,7 +13047,7 @@ int skill_check_condition_castbegin(struct map_session_data* sd, uint16 skill_id
 			sd->itemid = sd->itemindex = -1;
 			if( skill_id == WZ_EARTHSPIKE && sc && sc->data[SC_EARTHSCROLL] && rnd()%100 > sc->data[SC_EARTHSCROLL]->val2 ) // [marquis007]
 				; //Do not consume item.
-			else if( sd->status.inventory[i].expire_time == 0 ) // Rental usable items are not consumed until expiration
+			else if( sd->status.inventory[i].expire_time == 0 && sd->inventory_data[i]->flag.keepafteruse) // Rental usable items are not consumed until expiration
```

Comment:
being deleted ? a consumable rental should never exist so, if you wanna keep it use the flag.